### ISSUE TITLE
Try hints in consistency expressions

### ISSUE THREAD

**Issue by jgraley** [jgraley/inferno-cpp2v#469]
Firstly, make the `CHECK_HINTS` code in `SymbolicConstraint::Test()` permanent, but don't assert, just fail without hint if the hint doesn't re-evaluate as true.

This is because the solutions we get from solver are "sunny day" solutions and only expected to be valid if a valid solution exists.

<strike>Then add to `SymSolver` a function that substitutes solution for target in the original equation, making a "check equation" - this should be generated during planning (for each target) and `SymbolicConstraint::Test()` should try to evaluate that before the hint expression.

It's likely that after the substitution, some simplification will be possible. Also `SymbolicConstraint::Test()` should not now need to take a copy of the assignments in order to modify it.</strike> no, see comment

Do after #467 which provides a canary for this.

**Comment by jgraley** [comment]
Not so sure about this - don't want to modify the consistency expression in-place and would rather avoid adding cloners to these classes.
